### ISSUE TITLE
Why is my initial learning rate value being ignored by YOLOv8?

### ISSUE THREAD

**Issue by AndreaPi** [ultralytics/ultralytics#3629]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I want to change the initial learning rate for my model. I use the following train script to set an initial learning rate of 1e-1 (I know it's too large: this is just a test);

```
rom ultralytics import YOLO

experiment = 'yolov8_nano_bs8_300e_lr1e-1'
data_path = 'configs/dataset.yaml'
n_epochs = 300
bs = 8
n_workers = bs
gpu_id = 0
verbose = True
rng = 0
validate = True
patience = 0
base_lr = 1e-1
project = '../../logs/'

model = YOLO('yolov8n.pt')

results = model.train(
    data=data_path,
    epochs=n_epochs,
    batch=bs,
    device=gpu_id,
    verbose=verbose,
    seed=rng,
    val=validate,
    project=project,
    name=experiment,
    workers=n_workers,
    patience=patience,
    lr0=base_lr
)

```

When I start training, however, YOLO tells me that it's using an initial learning rate of 0.001429.  **Why is argument `lr0` being ignored?**

```
TensorBoard: Start with 'tensorboard --logdir ../../logs/yolov8_nano_bs8_300e_lr1e-2_test', view at http://localhost:6006/
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
AMP: checks passed ✅
train: Scanning /data/xx/xx/train/labels... 248 images, 0 backgrounds, 0 corrupt: 100%|██████████| 248/248 [00:00<00:00, 2274.71it/s]
train: WARNING ⚠️ /data/xx/xx/train/images/775ace8d-20220421_151317.jpeg: 9 duplicate labels removed
train: WARNING ⚠️ /data/xx/xx/train/images/b0e13878-image_1676460335000.jpeg: 8 duplicate labels removed
train: WARNING ⚠️ Cache directory /data/xx/xx/train is not writeable, cache not saved.
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))
val: Scanning /data/xx/xx/val/labels... 46 images, 0 backgrounds, 0 corrupt: 100%|██████████| 46/46 [00:00<00:00, 1669.71it/s]
val: WARNING ⚠️ Cache directory /data/xx/xx/val is not writeable, cache not saved.
Plotting labels to ../../logs/yolov8_nano_bs8_300e_lr1e-2_test/labels.jpg...
/opt/conda/lib/python3.10/site-packages/seaborn/axisgrid.py:118: UserWarning: The figure layout has changed to tight
  self._figure.tight_layout(*args, **kwargs)
optimizer: AdamW(lr=0.001429, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias(decay=0.0)
```

Environment:

```
Ubuntu 18.04.6 LTS
Python 3.10.9
>>> ultralytics.__version__
'8.0.131'
```

**Comment by bit-scientist** [comment]
hi @AndreaPi, as per #3360, if the dataset is small (you do have a small dataset: 248:46), `smartoptimizer()` sets the optimizer to use AdamW, hence the learning (`lr_fit`) is acquired by `round(0.002 * 5 / (4 + nc), 6)`.

**Comment by glenn-jocher** [comment]
Hi @AndreaPi! Thank you for reaching out. Based on the information provided, it seems that the initial learning rate (`lr0`) you specified is being ignored by YOLOv8.

In the YOLOv8 codebase, there is a function called `smartoptimizer()` that determines the optimizer to be used based on the size of the dataset. If the dataset is small, as in your case with 248 images in the training set, YOLOv8 automatically sets the optimizer to AdamW and calculates the learning rate (`lr_fit`) using the formula `round(0.002 * 5 / (4 + nc), 6)`. This is why you see the learning rate of 0.001429 being used instead of the value you specified.

I hope this explanation clarifies why the `lr0` argument is being ignored in your scenario. If you have any further questions or need assistance, please feel free to ask.

**Comment by glenn-jocher** [comment]
@AndreaPi hi there! Thank you for bringing this to our attention. We appreciate your feedback and understand how this can be confusing for new users.

We apologize for any confusion caused by the current documentation. To address your concern, we will update the documentation for the `lr0` argument in the training mode to explicitly mention that it may be ignored for small datasets. We'll also provide clarification on what we consider a "small" dataset.

Our goal is to provide clear and comprehensive documentation that helps users understand the behavior of the model and its parameters. Your feedback is valuable in improving the user experience, and we strive to make our documentation as informative as possible.

If you have any further questions or feedback, please don't hesitate to reach out. We're here to help. Thank you for bringing this to our attention, and we'll work to make the necessary updates to the documentation.

Best regards,
Glenn and the Ultralytics team
